rbs/prism/CommentsAssociatorPrism.cc

```diff
@@ -1,17 +1,1069 @@
 #include "rbs/prism/CommentsAssociatorPrism.h"
 
+#include "absl/strings/match.h"
+#include "absl/strings/str_split.h"
+#include "core/errors/rewriter.h"
+#include "parser/helper.h"
+#include "parser/prism/Helpers.h"
+#include "rbs/SignatureTranslator.h"
+
 using namespace std;
+using namespace sorbet::parser::Prism;
 
 namespace sorbet::rbs {
 
-CommentsAssociatorPrism::CommentsAssociatorPrism(core::MutableContext ctx, const parser::Prism::Parser &parser,
-                                                 const std::vector<core::LocOffsets> &commentLocations)
-    : ctx(ctx), parser(parser), commentLocations(commentLocations) {}
+const string_view CommentsAssociatorPrism::RBS_PREFIX = "#:";
+const string_view CommentsAssociatorPrism::ANNOTATION_PREFIX = "# @";
+const string_view CommentsAssociatorPrism::MULTILINE_RBS_PREFIX = "#|";
+const string_view CommentsAssociatorPrism::BIND_PREFIX = "#: self as ";
+
+namespace {
+
+// Static regex patterns to avoid recompilation
+const regex TYPE_ALIAS_PATTERN_PRISM("^#: type\\s*([a-z][A-Za-z0-9_]*)\\s*=\\s*([^\\n]*)$");
+const regex HEREDOC_PATTERN_PRISM("\\s*=?\\s*<<(-|~)[^,\\s\\n#]+(,\\s*<<(-|~)[^,\\s\\n#]+)*");
```

Comment:
Side-note: C++'s std::regex is [comically slow](https://stackoverflow.com/questions/70583395/why-is-stdregex-notoriously-much-slower-than-other-regular-expression-librarie). Famously, it's faster to [start up a PHP VM](https://cor3ntin.github.io/posts/abi/) to evaluate a regex, than it is to use `std::regex` directly.

We might be able to speed up the RBS rewriter by switching to another implementation, like:

  - [ctre](https://github.com/hanickadot/compile-time-regular-expressions)
  - Google's [RE2](https://github.com/google/re2)
  - [boost::regex](https://github.com/boostorg/regex)